GeoNode/geonode
# Developement using remote container

## Context
These changes allow us to develop inside a docker container.

## What was done
* A `.devcontainer/docker-compose.yml` was created, to override some instructions inside `docker-compose.yml`.
* A `.devcontainer/.env` also was created, to be used when developing inside the container.
* An error was throw when building docker compose because `docker-compose.yml` was defining `services` twice. Fixed.

## How to test/use
There are two options to develop inside the container, one using the [vscode "remote containers" extension](https://code.visualstudio.com/docs/remote/containers) (easier), other not using it and running by cmd line and any other code editor (usually harder).
* Option A - Using vscode remote containers:
    * Add the extension to your vscode: `ms-vscode-remote.remote-containers`
    * On your command pallet, select: "Remote-Containers: Reopen in Container"
    * In the first time it will take some time to build the images, then a vscode window will be opened connected to your container. 
    * You can use as if you are coding on your machine but in your vscode terminal you gonna see something similar to `root@77e80acc89b8:/usr/src/geonode# ` and the message `Dev Container: Debug Docker Compose` will appear in the bottom-left corner of your window.
* Option B - Without vscode remote containers/Editing files outside container.
    * Building: `docker-compose --project-name geonode -f docker-compose.yml -f .devcontainer/docker-compose.yml build`
    * Starting containers: `docker-compose --project-name geonode -f docker-compose.yml -f .devcontainer/docker-compose.yml up`
    * Running django in debug mode: `docker exec -it django4geonode bash -c "python manage.py runserver 0.0.0.0:8000"`
    * You can edit the files/save and see your modifications, you can also use `ipdb` to add breakpoints and debug your code.
    * You can also debug using debugpy alongside with vscode, for this you have to enable debugpy inside your django4geonode container: `pip install debugpy -t /tmp && python /tmp/debugpy --wait-for-client --listen 0.0.0.0:5678 manage.py runserver 0.0.0.0:8000 --nothreading --noreload"`


## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:


The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**
